Create mime type tibble for .drive (#19)

LucyMcGowan opened the pull request.

jennybc commented on R/aa_utils.R:
```diff
@@ -12,11 +12,22 @@ NULL
 .drive$base_url <- "https://www.googleapis.com"
+.drive$mimetypes <-
+  system.file("extdata", "mimetypes.csv", package = "googledrive") %>%
+  read.csv(stringsAsFactors = FALSE)
```
You don't mean to keep my toy example in here, do you? If not, these lines and the file they load should go away.

LucyMcGowan replied:
yikes no I didn't mean to keep that - I accidentally merged it in with my last commit 🙈

LucyMcGowan commented:
okay I think this is ready for another look @jennybc. I 
* updated the mime type tibble to join in the extensions from the `mime::mimemap` & incorporated that in the `drive_upload()` if the type was not specified. Right now, in terms of specifying type, I only let it be a document, spreadsheet, or presentation (as it was before, I didn't change it [per this comment](https://github.com/tidyverse/googledrive/issues/7#issuecomment-302839031)) 
* updated some testing according
* Noticed that you cannot change the publication status of "non-Google Drive" files (files without "application/vnd.google-apps." in their mimeType), which makes sense because google doesn't track revisions with these files and publishing is an action on a revision ID, I added an error accordingly if the file doesn't have a Google mimeType

jennybc commented:
This basically looks good to me. Do you want to deal with the conflicts then merge? I think you should remove the fiddling with, say, `drive_list()` tests from this PR, so this is really just about mime types. Then it should merge cleanly. The other will need to be revised in light of #34 anyway and should be a separate PR or commits.

Re: our awful testing situation. Maybe each file that hits the API should have the code necessary to create the structures at the top, inside `if (FALSE) { ... }`. Then it's clearly filed with the relevant tests and either of us can rerun it as necessary.
